Microsoft.Toolkit.Mvvm/ComponentModel/ObservableObject.cs

```diff
@@ -220,6 +239,21 @@ protected bool SetProperty<T>(Expression<Func<T>> propertyExpression, T newValue
         /// <param name="propertyName">(optional) The name of the property that changed.</param>
         /// <returns><see langword="true"/> if the property was changed, <see langword="false"/> otherwise.</returns>
         protected bool SetProperty<T>(Expression<Func<T>> propertyExpression, T newValue, IEqualityComparer<T> comparer, [CallerMemberName] string? propertyName = null)
+        {
+            return SetProperty(propertyExpression, newValue, comparer, out _, propertyName);
+        }
+
+        /// <summary>
+        /// Implements the shared logic for <see cref="SetProperty{T}(Expression{Func{T}},T,IEqualityComparer{T},string)"/>
+        /// </summary>
+        /// <typeparam name="T">The type of property to set.</typeparam>
+        /// <param name="propertyExpression">An <see cref="Expression{TDelegate}"/> returning the property to update.</param>
+        /// <param name="newValue">The property's value after the change occurred.</param>
+        /// <param name="comparer">The <see cref="IEqualityComparer{T}"/> instance to use to compare the input values.</param>
+        /// <param name="oldValue">The resulting initial value for the target property.</param>
+        /// <param name="propertyName">(optional) The name of the property that changed.</param>
+        /// <returns><see langword="true"/> if the property was changed, <see langword="false"/> otherwise.</returns>
+        private protected bool SetProperty<T>(Expression<Func<T>> propertyExpression, T newValue, IEqualityComparer<T> comparer, out T oldValue, [CallerMemberName] string? propertyName = null)
```

Comment:
Is there any specific reason why this is private protected? There seems to be some value in this overload for devs, where you actually have the old value.

Reply:
I basically introduced this method just to allow us to reuse some internal code without duplicating it - this method basically powers all the `Expression<Func<T>>` overloads for wrapping observable models. I figured it'd be best to keep this internal as the signature is a bit weird, and also developers would likely not need this as they'd have direct access to the old value anyway. Consider the standard example for this method (from the sample app):

```csharp
public class ObservableUser : ObservableObject
{
    private readonly User user;

    public ObservableUser(User user) => this.user = user;

    public string Name
    {
        get => user.Name;
        set => SetProperty(() => user.Name, value);
    }
}
```

Here devs would already have direct access to the old property, it's just `user.Name`. We only need the overload to make the method signature easier to use for developers, so that they only need to pass the expression instead of both the expression and the previous value (which would likely also be more error prone).

Additionally, to maintain the simmetry with all the other helpers, I figured if this was visible for devs we'd then also need to expose the variant with the automatic `IEqualityComparer<T>` like with the other helpers? And I thought that would likely make the whole thing overkill, with too many helper methods for a potentially niche use case scenario? 🤔

Just sharing my thoughts, these were basically the reasons why I went with `private protected` in this case. Let me know what you think and if you think we should make any changes here!

Reply:
Got it. Makes sense!